### ISSUE TITLE
Yoigoshi Release

### ISSUE THREAD

**Issue by DoctorDiablo** [07th-mod/higurashi-console-arcs#37]
Translation is complete for Yoigoshi.  I'd like to get all the dll changes and other fixes out of the way before I do editing, since I do a full playthrough as part of editing and I can use that to identify any other issues along the way.  So here's a list of the tasks we need to complete before this release:

- [x] Run the compiler against the yoig_ and prol_ scripts to add adv/lipsync/etc support @enumag 
- [x] Merge [this pull request](https://github.com/07th-mod/higurashi-assembly/pull/6) into the console arcs branch in assembly @tellowkrinkle 
- [x] If there are any other dll changes being actively worked that can be finished in the next few days, try to get those done. (or just let me know that we're good to go with what we have now) @tellowkrinkle @idealpersona 
- [x] Compile the Assembly-CSharp.dll with the latest code changes @tellowkrinkle (or anyone else who knows how to do that)
- [x] ~Address the rest of the items identified in~ [~this pull request~](https://github.com/07th-mod/higurashi-console-arcs/pull/34) All items not yet finished are now in this list
- [x] Update the installer so that it includes [the new globalgamemanagers.assets file and HigurashiEp04.exe](https://github.com/07th-mod/higurashi-console-arcs/issues/3), which will change the text in the title bar and use a custom icon for the executable. @ItaloKnox or @drojf 
- [x] Implement the [tips unlock screen](https://github.com/07th-mod/higurashi-console-arcs/issues/32) in all 5 arcs @DoctorDiablo 
- [x] Clean up [this old dead code](https://github.com/07th-mod/higurashi-console-arcs/issues/33) @DoctorDiablo 
- [x] Incorporate the script changes [identified here](https://github.com/07th-mod/higurashi-console-arcs/issues/36) @DoctorDiablo Also, @idealpersona if you want to do anything more here, you're welcome to
- [x] Run through editing on the whole script @DoctorDiablo 
- [x] ~Investigate chapter jump implementation~ postponed until a later release
- [x] ~Menu animation~ postponed until a later release
- [x] Changes references to `GHimatsubushiDay` in Yoigoshi scripts to use `GConsoleChoice`
- [x] Add new scenario images (see the main arcs for examples)
- [x] Add video openings and call them in the scripts

To run the console arcs with the latest scripts, you'll need the new [CG](https://www.mediafire.com/file/ycdic6vppggnyfh/ConsoleArcs-CG.7z/file), [Voices](https://www.mediafire.com/file/x6t6oh29evfls6k/ConsoleArcs-Voices.7z/file), and [BGM](https://www.mediafire.com/file/hi6o70t4d8gkod8/ConsoleArcs-BGM.7z/file) archives, the new video archives ([Windows](https://www.mediafire.com/file/ewg93nap9dcyjag/ConsoleArcs-Movie.7z/file) / [Mac/Linux](https://www.mediafire.com/file/trgn99b40x9uens/ConsoleArcs-Movie_UNIX.7z/file)), plus the [latest version of the DLL](https://07th-mod.com/higurashi_dlls/consolearcs/Assembly-CSharp.dll).  It uses the same [SE](https://07th-mod.com/rikachama/ConsoleArcs-SE.7z), and UI packages ([Windows](https://07th-mod.com/rikachama/Himatsubushi-UI.7z) / [Mac/Linux](https://07th-mod.com/rikachama/Himatsubushi-UI_UNIX.7z)) as the previous version.

Feel free to add other items to the list if you think there's anything else that needs to be done before release.

**Comment by tellowkrinkle** [comment]
On the compilation one, if everything's working properly, you should get a compiled DLL of the latest version about 6 minutes after you merge the PR available from the link in the repo readme, status of that can be found by clicking on one of the `build passing` buttons.

**Comment by ItaloKnox** [comment]
@DoctorDiablo 
The best scenario for the installer or the current setup would be to not deal with that. We don't have any other (current) pack where the files you requested to be added can fit. Instead, I suggest you to simply add them to the releases you make following the default folder structure we use in all patches. If you do that, the installer will take care of this automatically, with the downside of these two files having to be included in every update you make.

If that's not good for you we can make a new pack called ``ConsoleArcs-System.7z`` or something like that. Your call, just let me know what you want to do.

**Comment by DoctorDiablo** [comment]
@ItaloKnox The issue there is that the files are slightly different based on your OS.  The globalgamemanagers file and the exe are just for Windows, and Macs use a .icns file and an updated Info.plist.  I guess the bundle could include all of those in the appropriate folders, the only harm is a small useless file taking up a little space.

**Comment by ItaloKnox** [comment]
Yeah, that could work. Can you upload the files in their respective folders for me?

**Comment by DoctorDiablo** [comment]
Here are the files in their respective folder structure:

[NewFiles-Mac.zip](https://github.com/07th-mod/higurashi-console-arcs/files/2715983/NewFiles-Mac.zip)
[NewFiles-Windows.zip](https://github.com/07th-mod/higurashi-console-arcs/files/2715984/NewFiles-Windows.zip)

**Comment by ItaloKnox** [comment]
[Just pushed an extra pack to the server](http://07th-mod.com/rikachama/ConsoleArcs-System.7z).

I will edit the wiki table to add it when you are ready to make the release.

**Comment by DoctorDiablo** [comment]
@tellowkrinkle Is there a way to get the build output for that pull request?  I need the dll to try out the chapter jump functionality.

**Comment by enumag** [comment]
Not automatically afaik. For your needs I think you would have to clone the repo, merge the PR branch into hima_mod and compile it yourself.

**Comment by ItaloKnox** [comment]
@DoctorDiablo Do you have a PSD with the console arcs scenario images and text? I need that to make the shortcuts screen for the chapter.

**Comment by tellowkrinkle** [comment]
@DoctorDiablo true true, here have a copy I compiled, should work with console arcs
[Assembly-CSharp.dll.zip](https://github.com/07th-mod/higurashi-console-arcs/files/2716828/Assembly-CSharp.dll.zip)

**Comment by DoctorDiablo** [comment]
I don't have the PSD for the console arcs scenario images, but @Grelo might.

**Comment by DoctorDiablo** [comment]
I went ahead and created scenario images of my own for Yoigoshi and Police Case Files.  They follow the same layer structure the main arcs are using.  The images are in the big CG archive in the first post.  This is what they look like in-game:

![image](https://user-images.githubusercontent.com/20390474/50617049-386e4480-0eb9-11e9-9256-d540cdc8961b.png)

![image](https://user-images.githubusercontent.com/20390474/50617113-6e132d80-0eb9-11e9-9320-45f424dbeb2b.png)

They're passable as-is, but I'm not totally satisfied with them.  If anyone is interested in tweaking them, feel free.

**Comment by idealpersona** [comment]
This is the first time the console arcs has movies i think?  I'm updating everything manually on top of my previous console arcs installation, and I did not have the AVProVideo.dll in the Plugins folder, so had to manually copy it over when setting everything up.  Is there anything we need to do to make sure the installer installs the plugin like with the other arcs?

**Comment by DoctorDiablo** [comment]
Yep, movies are new with this release.  I'm pretty sure I just need to add AVProVideo.dll in the appropriate folder in the main patch file, and the installer will take care of it.

**Comment by idealpersona** [comment]
Extended the rain BGM file "ame"
[ame-extended.zip](https://github.com/07th-mod/higurashi-console-arcs/files/2725270/ame-extended.zip)

**Comment by idealpersona** [comment]
Cancel that last message - instead of extending, I just modified some offending files in the BGM folder like "ame".  These short files had a brief silent period in the beginning which made looping extra ugly.  I modified them so that they don't have that pause and so that they just loop more naturally without too-noticeable hiccups.

[loop-modified-BGMs.zip](https://github.com/07th-mod/higurashi-console-arcs/files/2725364/loop-modified-BGMs.zip)

The files I modified are:
ame
futousurunabe
honou
kawa
mati
rein

There were plenty others, but unfortunately the BGM folder appears to be filled with sound effects too for some reason, so these were the only ones that seemed like they were BGMs meant to repeat.   let me know if there are more.  I tested the smooth repetition for all these in my DAW and in my music player, and only tested "ame" in-game, but it sounded good from what i heard; the game seemed to smoothly loop everything just like my music player.

**Comment by DoctorDiablo** [comment]
@enumag Do you know what this "Menu animation" thing is?  It was mentioned in the old pull request.

**Comment by enumag** [comment]
Yeah I wanted to animate the Hinamizawa background using the separate transparent images from PS3. But I don't have time to finish it at the moment and it's by no means necessary. Just ignore it.

**Comment by DoctorDiablo** [comment]
@idealpersona Can you fix the looping on one more audio file?  Filename is addse10

**Comment by idealpersona** [comment]
@DoctorDiablo 
[addse10.zip](https://github.com/07th-mod/higurashi-console-arcs/files/2731184/addse10.zip)

**Comment by enumag** [comment]
Looking at the [flowchart](https://ibb.co/dTRmmb) I'm pretty sure the next arc will be Tokihogushi.

**Comment by DoctorDiablo** [comment]
Yeah, next is Tokihogushi.  I actually did a little work on that arc before deciding to do Yoigoshi instead, so that one should take less time to complete.

And yes, the plan is to do all of them eventually.

**Comment by DoctorDiablo** [comment]
Released!
